server/app/src/main/java/com/google/heartrate/wearos/app/MainActivity.java

```diff
@@ -4,18 +4,81 @@
 import android.support.wearable.activity.WearableActivity;
 import android.widget.TextView;
 
-public class MainActivity extends WearableActivity {
+import com.google.heartrate.wearos.app.bluetooth.server.BluetoothServer;
+import com.google.heartrate.wearos.app.bluetooth.server.handlers.HeartRateServiceRequestHandler;
+import com.google.heartrate.wearos.app.gatt.GattException;
+import com.google.heartrate.wearos.app.gatt.heartrate.service.HeartRateGattService;
+import com.google.heartrate.wearos.app.sensors.HeartRateSensorListener;
+import com.google.heartrate.wearos.app.sensors.HeartRateValueSubscriber;
 
+
+/**
+ * Application main activity sets up heart rate server and show current heart rate.
+ *
+ * <p>Used only for testing {@link BluetoothServer} with {@link HeartRateGattService} hosted.
+ */
+public class MainActivity extends WearableActivity implements HeartRateValueSubscriber {
+
+    /** {@link TextView} to show current heart rate. */
     private TextView mTextView;
 
+    /** {@link BluetoothServer} for heart rate service hosting. */
+    private BluetoothServer bluetoothServer;
+
+    /** Sensor listener to get heart rate. */
+    private HeartRateSensorListener heartRateSensorListener;
+
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_main);
+        setAmbientEnabled();
+        mTextView = findViewById(R.id.text);
 
-        mTextView = (TextView) findViewById(R.id.text);
+        startHeartRateServer();
+    }
 
-        // Enables Always-on
-        setAmbientEnabled();
+    /**
+     * Start server with heart rate service hosted.
+     */
+    private void startHeartRateServer() {
+        try {
+            heartRateSensorListener = new HeartRateSensorListener(this);
+            heartRateSensorListener.registerSubscriber(this);
+
+            bluetoothServer = new BluetoothServer(this);
```

Comment:
Same for the bluetooth server, why not construct it in the service?

Reply:
See next PR